### PR TITLE
[SPARK-23383][Build][Minor]Make a distribution should exit with usage while detecting wrong options

### PR DESCRIPTION
**Pull request by yaooqinn**
## What changes were proposed in this pull request?
```shell
./dev/make-distribution.sh --name ne-1.0.0-SNAPSHOT xyz --tgz  -Phadoop-2.7
+++ dirname ./dev/make-distribution.sh
++ cd ./dev/..
++ pwd
+ SPARK_HOME=/Users/Kent/Documents/spark
+ DISTDIR=/Users/Kent/Documents/spark/dist
+ MAKE_TGZ=false
+ MAKE_PIP=false
+ MAKE_R=false
+ NAME=none
+ MVN=/Users/Kent/Documents/spark/build/mvn
+ ((  5  ))
+ case $1 in
+ NAME=ne-1.0.0-SNAPSHOT
+ shift
+ shift
+ ((  3  ))
+ case $1 in
+ break
+ '[' -z /Users/Kent/.jenv/candidates/java/current ']'
+ '[' -z /Users/Kent/.jenv/candidates/java/current ']'
++ command -v git
+ '[' /usr/local/bin/git ']'
++ git rev-parse --short HEAD
+ GITREV=98ea6a7
+ '[' '!' -z 98ea6a7 ']'
+ GITREVSTRING=' (git revision 98ea6a7)'
+ unset GITREV
++ command -v /Users/Kent/Documents/spark/build/mvn
+ '[' '!' /Users/Kent/Documents/spark/build/mvn ']'
++ /Users/Kent/Documents/spark/build/mvn help:evaluate -Dexpression=project.version xyz --tgz -Phadoop-2.7
++ grep -v INFO
++ tail -n 1
+ VERSION=' -X,--debug                             Produce execution debug output'
```
It is better to declare the mistakes and exit with usage than `break`

## How was this patch tested?

manually 

cc @srowen

### CODE REVIEW

**Review by srowen — APPROVED** [review]

File: dev/make-distribution.sh [hunk 1]
```diff
@@ -72,8 +76,15 @@ while (( "$#" )); do
     --help)
       exit_with_usage
       ;;
+    --*)
```
[the remark this answers is not in the document]
Comment (yaooqinn) [reply dev/make-distribution.sh]: I guess both options are ok too.

**Review by srowen — COMMENTED** [review]
Hm, help me understand the logic here.

Before, anything unrecognized was silently ignored.

Now, an unrecognized --option generates an error but continues. An unrecognized -option is silently ignored. Anything else unrecognized fails.

Why not just warn and fail on anything unrecognized?

File: dev/make-distribution.sh [hunk 2]
```diff
@@ -72,9 +76,19 @@ while (( "$#" )); do
     --help)
       exit_with_usage
       ;;
-    *)
+    --*)
+      echo "Error: $1 is not supported, it will be ignored and not take effect"
+      usage
+      shift
+      continue
```
Comment (felixcheung) [inline dev/make-distribution.sh]: +1 on not continue

**Comment by srowen** [comment]
Ping @yaooqinn

**Comment by yaooqinn** [comment]
sorry for not replying in time. the logic here is that firstly unrecognized - -options show warning message and usage information，secondly the - options end the while loop and treat the current one and following options as maven parameters which will be handled later by maven，and finally all other unrecognized options fail the process immediately.

**Comment by srowen** [comment]
Yes @yaooqinn, it's easy enough to read the logic, but, my question is _why_ treat these cases differently? Why not go all the way to failing on any unexpected input?

**Comment by yaooqinn** [comment]
OK， i will push a commit very soon

**Comment by kiszk** [comment]
retest this please

**Review by srowen — COMMENTED** [review]
I'm still not clear on why options with a single dash are always ignored? Are they args passed through to something else?

**Comment by yaooqinn** [comment]
yes， single dash means the coming ones are all maven options，they will be handled later by mvn command. we do not parse each single option in this while loop from then till now.

**Review by srowen — APPROVED** [review]
Got it. I suppose the Maven options will always then start with one dash, not zero or two, so this works for the foreseeable future.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**